### PR TITLE
Fix `IDE0054` for increment/decrement when dealing with user-defined operators

### PR DESCRIPTION
**Pull request by DoctorKrolic**
Fixes: https://github.com/dotnet/roslyn/issues/70651

### CODE REVIEW

File: src/Analyzers/CSharp/Tests/UseCompoundAssignment/UseCompoundAssignmentTests.cs [hunk 1]
```diff
@@ -1254,7 +1254,7 @@ void M()
                 """);
         }
 
-        [Fact, WorkItem(38054, "https://github.com/dotnet/roslyn/issues/53969")]
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/53969")]
```
Comment (DoctorKrolic) [inline src/Analyzers/CSharp/Tests/UseCompoundAssignment/UseCompoundAssignmentTests.cs:1257]: These are unrelated to the main change. I just noticed the mismatch while working and thought that it is too small change for a separate PR

**Comment by CyrusNajmabadi** [comment]
Please ping me next Monday for review

**Comment by DoctorKrolic** [comment]
@CyrusNajmabadi PTAL

**Review by CyrusNajmabadi — APPROVED** [review]

**Comment by CyrusNajmabadi** [comment]
Can you merge in lates main, then ping me?  I'll set to auto-merge then.

**Comment by DoctorKrolic** [comment]
@CyrusNajmabadi Restart failed pipelines pls
